src/components/EditionModal/EditionModal.jsx

```diff
@@ -0,0 +1,167 @@
+          onCancel={choosing.onCancel}
+        />
+      ) : null}
+      {choosing.type === CHOOSING_TYPES.accountOrGroup ? (
+        <AccountGroupChoice
+          current={choosing.value}
+          onSelect={choosing.onSelect}
+          onCancel={choosing.onCancel}
+        />
+      ) : null}
+    </>
+  )
+}
+
+const InfoSlide = ({
+  doc,
+  fieldSpecs,
+  fieldOrder,
+  fieldLabels,
+  onRequestChooseField,
+  onChangeField
+}) => {
+  return (
+    <ModalSections>
+      {fieldOrder.map(fieldName => {
+        const FieldSection = SectionsPerType[fieldSpecs[fieldName].type]
+        const fieldSpec = fieldSpecs[fieldName]
+        const fieldLabel = fieldLabels[fieldName]
+        return (
+          <FieldSection
+            key={fieldName}
+            label={fieldLabel}
+            value={fieldSpec.getValue(doc)}
+            onClick={
+              fieldSpec.immediate ? null : () => onRequestChooseField(fieldName)
+            }
+            onChange={
+              fieldSpec.immediate
+                ? ev => {
+                    onChangeField(fieldName, ev.target.value)
+                  }
+                : null
+            }
+          />
+        )
+      })}
+    </ModalSections>
+  )
+}
+
+const EditionModal = props => {
+  const {
+    fieldSpecs,
+    fieldLabels,
+    fieldOrder,
+    initialDoc,
+    modalTitle,
+    okButtonLabel,
+    cancelButtonLabel,
+    onEdit,
+    onDismiss
+  } = props
+  const [doc, setDoc] = useState(initialDoc)
+  const [choosing, setChoosing] = useState(null)
+
+  const handleChoosingCancel = () => {
+    setChoosing(null)
+  }
+
+  const handleChangeField = (name, val) => {
+    const updater = fieldSpecs[name].updater
+    const updatedDoc = updater(doc, val)
+    setDoc(updatedDoc)
+  }
+
+  const handleRequestChooseField = name => {
+    const fieldSpec = fieldSpecs[name]
+    setChoosing({
+      type: fieldSpec.type,
+      value: fieldSpec.getValue(doc),
+      onSelect: val => {
+        setChoosing(null)
+        handleChangeField(name, val)
+      },
+      onCancel: handleChoosingCancel
+    })
+  }
+
+  const handleConfirmEdit = () => {
+    onEdit(doc)
+  }
+
+  return (
+    <Modal title={modalTitle} mobileFullscreen={true} dismissAction={onDismiss}>
+      <Stepper
+        showPercentage={false}
+        currentIndex={choosing ? 1 : 0}
+        onBack={() => setChoosing(null)}
+      >
+        <InfoSlide
+          doc={doc}
+          fieldOrder={fieldOrder}
+          fieldSpecs={fieldSpecs}
+          fieldLabels={fieldLabels}
+          onRequestChooseField={handleRequestChooseField}
+          onChangeField={handleChangeField}
+        />
+        <div>{choosing ? <ChoosingSwitch choosing={choosing} /> : null}</div>
+      </Stepper>
+      <ModalFooter>
+        <ModalButtons>
+          <Button
+            theme={'secondary'}
+            onClick={onDismiss}
+            label={cancelButtonLabel(props, doc)}
+          />{' '}
```

Comment:
This space should be replaced by a CSS margin

Reply:
Removed, it was not used actually :)